src/mount/mastercomm.cc

```diff
@@ -476,261 +491,461 @@ void fs_register_config() {
-		put32bit(&wptr,pleng);
-		memcpy(wptr,gInitParams.subfolder.c_str(),pleng);
+		put32bit(&messageToMaster, subfolderPathLength);
+		memcpy(messageToMaster, gInitParams.subfolder.c_str(), subfolderPathLength);
 	}
+
 	if (havepassword) {
-		memcpy(wptr+pleng,digest,16);
+		memcpy(messageToMaster + subfolderPathLength, passwordDigest.data(), passwordDigestLength);
 	}
-	if (tcptowrite(fd,regbuff,8+64+(gInitParams.meta?9:13)+ileng+pleng+(havepassword?16:0),1000)!=(int32_t)(8+64+(gInitParams.meta?9:13)+ileng+pleng+(havepassword?16:0))) {
+
+	if (tcptowrite(fd, registrationMessageBuffer.data(), fuseRegisterNewSessionRequestLength,
+	               kDefaultTcpCommTimeoutMSeconds) !=
+	    (int32_t)(fuseRegisterNewSessionRequestLength)) {
 		if (verbose) {
-			fprintf(stderr,"error sending data to sfsmaster: %s\n",strerr(tcpgetlasterror()));
+			fprintf(stderr, "error sending data to sfsmaster: %s\n", strerr(tcpgetlasterror()));
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"error sending data to sfsmaster: %s",strerr(tcpgetlasterror()));
+			safs::log_warn("error sending data to sfsmaster: {}", strerr(tcpgetlasterror()));
 		}
 		tcpclose(fd);
-		fd=-1;
-		free(regbuff);
+		fd = -1;
 		return -1;
 	}
-	if (tcptoread(fd,regbuff,8,1000)!=8) {
+
+	if (tcptoread(fd, registrationMessageBuffer.data(), registerTotalHeaderInfoLength,
+	              kDefaultTcpCommTimeoutMSeconds) != (int32_t)(registerTotalHeaderInfoLength)) {
 		int tcplasterr = tcpgetlasterror();
-		const auto* errorMessage = (tcplasterr != 0) ? strerr(tcplasterr) : strerr(TCPNORESPONSE);
+		const auto *errorMessage = (tcplasterr != 0) ? strerr(tcplasterr) : strerr(TCPNORESPONSE);
 		if (verbose) {
-			fprintf(stderr,"error receiving data from sfsmaster: %s\n", errorMessage);
+			fprintf(stderr, "error receiving data from sfsmaster: %s\n", errorMessage);
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"error receiving data from sfsmaster: %s", errorMessage);
+			safs::log_warn("error receiving data from sfsmaster: {}", errorMessage);
 		}
 		tcpclose(fd);
-		fd=-1;
-		free(regbuff);
+		fd = -1;
 		return -1;
 	}
-	rptr = regbuff;
-	get32bit(&rptr, i);
-	if (i!=MATOCL_FUSE_REGISTER) {
+
+	messageFromMaster = registrationMessageBuffer.data();
+	get32bit(&messageFromMaster, messageValueIterator);
+	if (messageValueIterator != MATOCL_FUSE_REGISTER) {
 		if (verbose) {
-			fprintf(stderr,"got incorrect answer from sfsmaster\n");
+			fprintf(stderr, "got incorrect answer from sfsmaster\n");
 		} else {
-			safs_pretty_syslog(LOG_WARNING,"got incorrect answer from sfsmaster");
+			safs::log_warn("got incorrect answer from sfsmaster");
 		}
 		tcpclose(fd);
-		fd=-1;
-		free(regbuff);
+		fd = -1;
 		return -1;
 	}
-	get32bit(&rptr, i);
-	if (!(i==1 || (gInitParams.meta && (i==5 || i==9 || i==19)) || (gInitParams.meta==0 && (i==13 || i==21 || i==25 || i==35)))) {
+
+	get32bit(&messageFromMaster, messageValueIterator);
+	bool validNewMetaSessionDataLegth =
+	    kNewMetaSessionDataLengthLegacy ||
```

Comment:
I think this line should be:

```suggestion
	    messageValueIterator == kNewMetaSessionDataLengthLegacy ||
```

Reply:
Nice catch!